### PR TITLE
DEV-1247: Case Sensitive Username Fix

### PR DESCRIPTION
**Pull request by andy-t-wang**
Usernames currently allows case sensitive duplicates this is to prevent that.

closes DEV-1247

### CODE REVIEW

File: migrations/20241230195405_add_case_insensitive_username_indexes.sql [hunk 1]
```diff
@@ -0,0 +1,4 @@
+-- Add migration script here
+CREATE UNIQUE INDEX names_username_lower_idx ON names (LOWER(username));
+
+CREATE UNIQUE INDEX old_names_username_lower_idx ON old_names (LOWER(old_username));
```
Comment (paul-schaaf) [inline migrations/20241230195405_add_case_insensitive_username_indexes.sql:4]: this will fail if we already have non-unique usernames wont it?

**Review by paul-schaaf — APPROVED** [review]

File: migrations/20241230195405_add_case_insensitive_username_indexes.sql (hunk 1 above)
Comment (andy-t-wang) [reply migrations/20241230195405_add_case_insensitive_username_indexes.sql:4]: ah good catch let me see

Comment (andy-t-wang) [reply migrations/20241230195405_add_case_insensitive_username_indexes.sql:4]: Will wait to merge this till we run the dedup script on the usernames table

Comment (andy-t-wang) [reply migrations/20241230195405_add_case_insensitive_username_indexes.sql]: actually I just removed the unique constraint. This is fine there wont be too many duplicates anyways

**Pull request merged**